### PR TITLE
BAN-115: Add test cases to tooltip and Notifications

### PR DESCRIPTION
**Pull request by mihaelpalfi**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/BAN-115)

#### Description
As per JIRA ticket

#### ChangeLOG

- [ ] I have added all notable deployment/development environment (onprem!) changes to [CHANGELOG.md](https://github.com/reciprocity/zen-ui/blob/main/README.md)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
-        <mock:shadow-root>
-          <slot></slot>
-          <div class="arrow dark top"></div>
-        </mock:shadow-root>
-      </zen-tooltip>
-    `);
+    expect(page.root.getAttribute('variant')).toEqual(variant);
+  });
+
+  it.each(['top', 'right', 'bottom', 'left'])('Test each position %s is displayed correct', async position => {
```
Comment (chavlji) [inline src/components/zen-tooltip/test/zen-tooltip.spec.tsx:13]: nit: Test name should be `applies corret position attribute is %s`

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx [hunk 2]
```diff
@@ -1,19 +1,44 @@
...
+    const page = await newSpecPage({
+      components: [ZenTooltip],
+      html: `<zen-tooltip position="${position}" label="Testing tooltip"></zen-tooltip>`,
+    });
+    expect(page.root.getAttribute('position')).toEqual(position);
```
Comment (chavlji) [inline src/components/zen-tooltip/test/zen-tooltip.spec.tsx:18]: This test only checks if Stencil is applying the correct attribute. We should test our component.

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx [hunk 3]
```diff
@@ -1,19 +1,44 @@
...
     });
-    expect(page.root).toEqualHtml(`
-      <zen-tooltip class="dark tooltip top" style="max-height: none;">
-        <mock:shadow-root>
-          <slot></slot>
-          <div class="arrow dark top"></div>
-        </mock:shadow-root>
-      </zen-tooltip>
-    `);
+    expect(page.root.getAttribute('variant')).toEqual(variant);
```
Comment (chavlji) [inline src/components/zen-tooltip/test/zen-tooltip.spec.tsx:10]: Same as for position test. This test isn't very useful.

File: src/components/helpers/types.ts [hunk 4]
```diff
@@ -1,11 +1,14 @@
 // Props:
-
 export type Align = 'left' | 'right' | 'center';
 
 export type Position = 'top' | 'right' | 'bottom' | 'left';
 
 export type TooltipVariant = 'dark' | 'light' | 'error';
 
+export type NotificationVariant = 'success' | 'info' | 'warning' | 'error';
```
Comment (chavlji) [inline src/components/helpers/types.ts:8]: Nice!

**Review by chavlji — CHANGES_REQUESTED** [review]

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx (hunk 1 above)
Comment (mihaelpalfi) [reply src/components/zen-tooltip/test/zen-tooltip.spec.tsx]: Changed

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx (hunk 2 above)
Comment (mihaelpalfi) [reply src/components/zen-tooltip/test/zen-tooltip.spec.tsx]: You are right. Changed that it checks if correct class is applied.

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx (hunk 3 above)
Comment (mihaelpalfi) [reply src/components/zen-tooltip/test/zen-tooltip.spec.tsx]: Same as above.

File: src/components/helpers/types.ts (hunk 4 above)
Comment (mihaelpalfi) [reply src/components/helpers/types.ts:8]: Still think that here should be an enum :)

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx [hunk 5]
```diff
@@ -7,16 +7,24 @@ describe('Test parameters rendering', () => {
...
+      const page = await newSpecPage({
+        components: [ZenTooltip],
+        html: `<div>Trigger</div><zen-tooltip position="${position}" always-visible="true" label="Testing tooltip"></zen-tooltip>`,
+      });
+
+      // We have to wait that tooltip is displayed because of always visible param
+      setTimeout(async () => {
+        expect(page.root.classList.contains(position)).toBe(true);
+        done();
+      }, 101);
```
Comment (chavlji) [inline src/components/zen-tooltip/test/zen-tooltip.spec.tsx:25]: Why 101? Doesn't it work if we lower that to eg. 10? Imo the point is just to postpone it to next cycle.

**Review by chavlji — APPROVED** [review]

File: src/components/helpers/types.ts (hunk 4 above)
Comment (chavlji) [reply src/components/helpers/types.ts:8]: Agree with that. I'll take another look into that.

File: src/components/zen-tooltip/test/zen-tooltip.spec.tsx (hunk 5 above)
Comment (mihaelpalfi) [reply src/components/zen-tooltip/test/zen-tooltip.spec.tsx:25]: Because somebody implemented that it should wait 100ms before showing :)

**Pull request merged**